### PR TITLE
WAL Compression enhancement

### PR DESCRIPTION
**Pull request by OneSizeFitsQuorum**
1. enbale wal compression default
2. enhance some codes and comments

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/LogWriter.java [hunk 1]
```diff
@@ -49,37 +50,36 @@ public abstract class LogWriter implements ILogWriter {
   protected final FileChannel logChannel;
   protected long size = 0;
   protected long originalSize = 0;
-  private final ByteBuffer headerBuffer = ByteBuffer.allocate(Integer.BYTES * 2 + 1);
+  private final int COMPRESSED_HEADER_SIZE = Byte.BYTES + Integer.BYTES * 2;
+  private final int UN_COMPRESSED_HEADER_SIZE = Byte.BYTES + Integer.BYTES;
```
Comment (jt2594838) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/LogWriter.java:54]: Better with some comments.

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/LogWriter.java [hunk 2]
```diff
@@ -49,37 +50,36 @@ public abstract class LogWriter implements ILogWriter {
...
   private ICompressor compressor =
       ICompressor.getCompressor(
           IoTDBDescriptor.getInstance().getConfig().getWALCompressionAlgorithm());
   private ByteBuffer compressedByteBuffer;
-  // Minimum size to compress, default is 32 KB
-  private static long minCompressionSize = 32 * 1024L;
 
-  protected LogWriter(File logFile) throws IOException {
+  /** Minimum size to compress, default is 32 KB */
+  private static long MIN_COMPRESSION_SIZE = 32 * 1024L;
```
Comment (jt2594838) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/LogWriter.java:62]: It seems unconfigurable, then no need to say "default".

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALFileVersion.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      channel.position(0);
+      ByteBuffer buffer = ByteBuffer.allocate(WALWriter.MAGIC_STRING_V2_BYTES);
+      channel.read(buffer);
+      buffer.flip();
+      if (buffer.remaining() < WALWriter.MAGIC_STRING_V2_BYTES) {
+        return UNKNOWN;
+      }
+      return new String(buffer.array(), StandardCharsets.UTF_8).equals(WALWriter.MAGIC_STRING_V2)
+          ? V2
+          : V1;
```
Comment (jt2594838) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALFileVersion.java:50]: Is it possible that the string still does not equal to both V1 and V2?

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALInputStream.java [hunk 4]
```diff
@@ -38,11 +39,12 @@ public class WALInputStream extends InputStream implements AutoCloseable {
 
   private static final Logger logger = LoggerFactory.getLogger(WALInputStream.class);
   private final FileChannel channel;
-  private final ByteBuffer segmentHeaderBuffer = ByteBuffer.allocate(Integer.BYTES + Byte.BYTES);
-  private final ByteBuffer compressedHeader = ByteBuffer.allocate(Integer.BYTES);
+  private final ByteBuffer segmentHeaderWithoutCompressedSizeBuffer =
+      ByteBuffer.allocate(Integer.BYTES + Byte.BYTES);
```
Comment (jt2594838) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALInputStream.java:43]: Better with some comments.

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALInputStream.java [hunk 5]
```diff
@@ -38,11 +39,12 @@ public class WALInputStream extends InputStream implements AutoCloseable {
...
+  private final ByteBuffer compressedSizeHeaderBuffer = ByteBuffer.allocate(Integer.BYTES);
```
Comment (jt2594838) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALInputStream.java:44]: I think "compressedSizeBuffer" is enough. The "header" is confusing.

**Review by jt2594838 — APPROVED** [review]

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/LogWriter.java (hunk 2 above)
Comment (OneSizeFitsQuorum) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/LogWriter.java:62]: Fixed

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/LogWriter.java (hunk 1 above)
Comment (OneSizeFitsQuorum) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/LogWriter.java:54]: Fixed

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALFileVersion.java (hunk 3 above)
Comment (OneSizeFitsQuorum) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALFileVersion.java:50]: Normally not, unless someone tampers with the file

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALInputStream.java (hunk 5 above)
Comment (OneSizeFitsQuorum) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALInputStream.java:44]: Fixed

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALInputStream.java (hunk 4 above)
Comment (OneSizeFitsQuorum) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALInputStream.java:43]: Fixed

File: .github/workflows/cluster-it-1c1d.yml
Comment (HTHou) [inline .github/workflows/cluster-it-1c1d.yml:1]: Mistakenly remove these workflows files?

**Pull request merged**